Cleanup

zcawood333 opened the pull request.

zcawood333 committed:
upload requests create readstream when initialized

zcawood333 committed:
simplified encrypted uploads
removed temp files and aesEncrypt(), and simplified httpUpload() and initMulticastClient()

zcawood333 committed:
combined PUT and POST into upload

zcawood333 committed:
renamed GET() to get()

zcawood333 committed:
reorganized funcs

zcawood333 committed:
simplified downloadFileName assignment

zcawood333 committed:
downloadFileName changed to downloadFilePath

zcawood333 committed:
changed getDownloadPath to downloadPath

zcawood333 committed:
GetRequest creates a decrypting writeStream at init

zcawood333 committed:
simplified encrypted downloads
removed temp files and aesDecrypt(), and simplified initRequest()

zcawood333 committed:
removed validateDirPath()

zcawood333 committed:
renamed checkFilekey to checkFileKey

zcawood333 committed:
removed old flag-based implementation

zcawood333 committed:
removed work in progress from top

zcawood333 committed:
renamed downloadPath to downloadDir and took off trailing /

zcawood333 committed:
cleanup

zcawood333 committed:
fixed 'encrypted' label on unencrypted GET

zcawood333 committed:
remove files created by writeStream if not written to

zcawood333 committed:
fixed bug: request failed immediately after last message
request should fail after last message has been sent and had time to be processed

zcawood333 committed:
initial commit

zcawood333 committed:
ignore .sh files

zcawood333 committed:
added PUT/POST encrypted bucket uploads

zcawood333 committed:
removed old implementation tests

zcawood333 committed:
renamed 'node_proj' to 'server'

zcawood333 committed:
put _ before private methods

zcawood333 committed:
used this.end() rather than manually ending

zcawood333 committed:
random cleanup

zcawood333 committed:
changed ' to "

zcawood333 committed:
changed ' to "

zcawood333 committed:
changed ' to "

zcawood333 committed:
random cleanup

zcawood333 committed:
revised console.log statements

zcawood333 committed:
added function summaries

zcawood333 committed:
added and implemented callbacks for validateLogFile()

zcawood333 committed:
added function summaries

zcawood333 committed:
made function summaries readable via hover

horner commented on .vscode/launch.json:
```diff
@@ -59,7 +59,7 @@
         },
         {
-            "name": "Client GET encrypted (multiple output files w/ bad output file)",
+            "name": "Client GET (multiple output files w/ bad output file)",
```
Why did you removed encrypted?

zcawood333 replied:
because it was a typo

horner commented on client/.gitignore:
```diff
@@ -2,4 +2,5 @@ node_modules/
 downloads/
 tmp/
 logs/
```
```suggestion
logs/*.tmp
```

horner committed:
Update README.md

zcawood333 committed:
removed POST

zcawood333 committed:
undefined bucket uses default, not throw error

zcawood333 committed:
removed POST and reorganized around PUT/GET

zcawood333 committed:
cleaner Object construction
reduced optional args, removed unnecessary attr ('arg') in base class

zcawood333 committed:
added fileKey attr to PutRequest

zcawood333 committed:
simplified funcs
sendMulticastMsg removed optional args and added a callback
logUpload/Download now use 'fileKey' attr instead of concatenating other attrs

zcawood333 committed:
condensed argv parsing and processing
all process args are now processed in processArgv

zcawood333 committed:
added arg attr back in for failure printing
prints 'FAILED: undefined' otherwise

zcawood333 committed:
random cleanup
removed bucketHandler
removed logging statements in processArgv
put app.get('/exist'...) in debug

horner committed:
Update README.md

zcawood333 committed:
initial commit

zcawood333 committed:
removed fileSize parameter

zcawood333 committed:
initial conversion from terminal based to library

zcawood333 committed:
recreated terminal functionality removed from client.js

zcawood333 committed:
calls cmdClient.js instead of client.js now

zcawood333 committed:
fixed bug caused by removing fileSize parameter

zcawood333 committed:
added eslint

zcawood333 committed:
Added ESlint

zcawood333 committed:
added babel and fixed linting errors

zcawood333 committed:
added dist

zcawood333 committed:
changed client/cmdClient to client/src/cmdClient

zcawood333 committed:
added callbacks to put/get

zcawood333 committed:
now automatically closes when finished

zcawood333 committed:
added promise attribute for async operation in client.js
